src/track/cueinfoimporter.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+/// cueinfoimporter.h
+/// Created 2020-04-24 by Jan Holthuis
+
+#include "track/cueinfo.h"
+
+namespace mixxx {
+
+/// Importer class for CueInfo objects that can correct timing offsets when the
+/// signal info (channel number, sample rate, bitrate) is known.
+class CueInfoImporter {
+  public:
+    CueInfoImporter();
+    CueInfoImporter(const QList<CueInfo>& cueInfos);
+
+    /// Returns audio signal dependent timing offset correction.
+    /// The default implementation just returns 0, but this can be overridden
+    /// in subclasses.
+    virtual double getTimingOffsetMillis(
+            const QString& filePath,
+            const audio::SignalInfo& signalInfo) const;
+
+    void append(const CueInfo& cueInfo);
+    void append(const QList<CueInfo>& cueInfos);
+
+    QList<CueInfo> getCueInfos() const;
```

Comment:
Do you need both functions?

In my opinion the importer should strictly only do 2 things:
- Collect new cue infos
- Convert the collected cue infos once, i.e. becoming empty after the conversion